eng/scripts/automation/GenerateAndBuildLib.ps1

```diff
@@ -43,43 +50,37 @@ function Update-AutorestConfigFile() {
         [string]$readme = ""
     )
     if (Test-Path -Path $autorestFilePath) {
-        if ($readme -ne "") {
-            Write-Host "Updating autorest.md file to config required readme file."
+        if (($readme -ne "") -or ($inputfile -ne "")) {
             $requirRex = "require*:*";
-            $inputfileRex = "input-file *:*"
-            $requirefile = $readme + [Environment]::NewLine + "- " + $readme.Replace("readme.md", "readme.csharp.md")
-            if ((Get-Content $autorestFilePath | Select-String -Pattern $requirRex).Matches.Success) {
-                (Get-Content $autorestFilePath) -notmatch "- .*.md" |Out-File $autorestFilePath
-                (Get-Content $autorestFilePath) -notmatch $inputfileRex |Out-File $autorestFilePath
-                (Get-Content $autorestFilePath) -notmatch "- .*.json" |Out-File $autorestFilePath
-                (Get-Content $autorestFilePath) -replace $requirRex, ("require:" + [Environment]::NewLine + "- " + "$requirefile") | Set-Content $autorestFilePath
-            } elseif ((Get-Content $autorestFilePath | Select-String -Pattern $inputfileRex).Matches.Success) {
-                (Get-Content $autorestFilePath) -notmatch "- .*.json" |Out-File $autorestFilePath
-                $requirefile = $requirefile + [Environment]::NewLine + "csharp: true";
-                (Get-Content $autorestFilePath) -replace $inputfileRex, ("require:" + [Environment]::NewLine + "- " + "$requirefile") | Set-Content $autorestFilePath
-            }
-            if ( $? -ne $True) {
-                Write-Error "Failed to update autorest.md. exit code: $?"
-                exit 1
-            }
-        } elseif ($inputfile -ne "") {
-            Write-Host "Updating autorest.md file to update input-file."
-            $inputfileRex = "input-file *:*"
-            if ((Get-Content $autorestFilePath | Select-String -Pattern $inputfileRex).Matches.Success) {
-                (Get-Content $autorestFilePath) -notmatch "- .*.json" |Out-File $autorestFilePath
-                (Get-Content $autorestFilePath) -replace $inputfileRex, ("input-file:" + [Environment]::NewLine + "$inputfile") | Set-Content $autorestFilePath
-            } else {
-                $startNum = (Get-Content $autorestFilePath | Select-String -Pattern '```').LineNumber[0]
-                $fileContent = Get-Content -Path $autorestFilePath
-                $fileContent[$startNum - 1] += ([Environment]::NewLine + "input-file:" + [Environment]::NewLine + "$inputfile")
-                $fileContent | Set-Content $autorestFilePath
+            $inputfileRex = "input-file*:*"
+            # clear
+            (Get-Content $autorestFilePath) -notmatch $requirRex |Out-File $autorestFilePath
```

Comment:
Why are we reading and writing this file multiple times as opposed to reading once and then working with the string in memory?

Reply:
Thanks @weshaggard  I will investigate if we can combine those all together and use common instance in memory to apply all those actions.